File: Java Programs/migratory_birds.java
import java.util.HashMap;
import java.util.Map;
import java.util.Scanner;

public class Main {

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String[] input = scanner.nextLine().split(" ");

        // Count the frequency of each plant ID
        Map<Integer, Integer> frequencyMap = new HashMap<>();
        for (String s : input) {
            int plantID = Integer.parseInt(s);
            frequencyMap.put(plantID, frequencyMap.getOrDefault(plantID, 0) + 1);
        }

        // Find the most common plant ID
        int mostCommonPlantID = Integer.MAX_VALUE;
        int maxFrequency = 0;
        for (Map.Entry<Integer, Integer> entry : frequencyMap.entrySet()) {
            int plantID = entry.getKey();
            int frequency = entry.getValue();
            if (frequency > maxFrequency || (frequency == maxFrequency && plantID < mostCommonPlantID)) {
                mostCommonPlantID = plantID;
                maxFrequency = frequency;
            }
        }

        // Output the most common plant ID
        System.out.println(mostCommonPlantID);
    }
}
                                                                                                                            

File: Python Programs/migratory_birds.py
from collections import Counter

def most_common_plant(sightings):
    # Count the occurrences of each plant ID
    plant_counts = Counter(sightings)
    
    # Sort the plant counts by frequency and plant ID
    sorted_plant_counts = sorted(plant_counts.items(), key=lambda x: (-x[1], x[0]))
    
    # Return the plant ID with the highest frequency
    return sorted_plant_counts[0][0]

# Take user input
sightings = list(map(int, input().split()))

# Find the most common plant ID and output the result
print(most_common_plant(sightings))
                                                                                                                            